irr/src/CIrrDeviceSDL.cpp

```diff
@@ -858,18 +884,31 @@ bool CIrrDeviceSDL::run()
 
 		case SDL_KEYDOWN:
 		case SDL_KEYUP: {
-			SKeyMap mp;
-			mp.SDLKey = SDL_event.key.keysym.sym;
-			s32 idx = KeyMap.binary_search(mp);
+			auto keysym = SDL_event.key.keysym.sym;
+			auto scancode = SDL_event.key.keysym.scancode;
+
+			// Treat AC_BACK as the Escape key
+			if (scancode == SDL_SCANCODE_AC_BACK || scancode == SDL_SCANCODE_ESCAPE)
+			{
+				if (SDL_event.type == SDL_KEYDOWN)
+					escapeKeys.insert(scancode);
+				else
+					escapeKeys.erase(scancode);
+				if (SDL_event.type == SDL_KEYUP && !escapeKeys.empty())
+					break; // avoid sending KEYUP twice if AC_BACK and ESCAPE are both released
+				scancode = SDL_SCANCODE_ESCAPE;
+				keysym = SDLK_ESCAPE;
+			}
 
+			const auto &entry = KeyMap.find(keysym);
 			EKEY_CODE key;
-			if (idx == -1)
-				key = (EKEY_CODE)0;
+			if (entry == KeyMap.end())
+				key = (EKEY_CODE)KEY_UNKNOWN;
```

Comment:
```suggestion
				key = KEY_UNKNOWN;
```

could also just make this a ternary